### DISCUSSION TITLE
코드리뷰 요청

### DISCUSSION THREAD

**Discussion by javascriptbiggosoo** [wanted-9th-3team/frontend-study#70] [프론트 기술 질문 공간] [answered]
https://github.com/javascriptbiggosoo/nomflix
검색기능과 TV 페이지, 모달 등 구현 중인 부분이 있지만 일단 올려봅니다

**Comment by michoball** [comment, marked as answer]
잘모르는 라이브러리에 대한 코드는 제외하고 리뷰드리겠습니다. 

main.tsx: react-query devtools도 같이 사용하시면 편합니다.  https://tanstack.com/query/v4/docs/react/devtools

App.tsx: router 를 따로 분리하신 점이 좋은 것 같습니다. 
              React.Fragmant를 빼도 될 것 같습니다. 

HomePage.tsx: apis/tmdb.ts의 fetch 함수에서 return type을 정의하고 사용하면  query 를 사용하는 쪽에서 타입을 정의할 필요가 없으니 
                        코드가 간결해질 것 같습니다. 
                        useMatch 함수는 사용해본 적이 없는데 이런게 있네요. 저도 나중에 사용해봐야겠습니다. 

SearchResultPage.tsx: 
              line 16 :  HomePage.tsx에서 사용하신 것처럼 data를 다른 이름으로 바꾸는게 가독성에 좋을 것 같습니다. 
              line 18 :  bind는 빼도 작동하지 않나요?
              line 30, 33, 39 : homePage에서와 동일한 이유로 filter나 map에서 타입을 정의하지 않는 것이 더 좋을 것 같습니다. 

useResize.ts:  useState에 width만 사용하는거면 객체로 상태를 가지지 않아도 될 것 같습니다. 

MovieModal.tsx:  이곳에 정의된 interface들이 apis/tmdb에 있고 fetch함수 사용시 return type으로 전달하는 것이 좋아보입니다.
                             utility type을 사용해서 중복되게 정의된 값을 줄이는게 좋아보입니다. 

Slider.tsx: [오른쪽이면1]= useState()가 어떤 역할을 하는지 state이름만으로는 잘 모르겠습니다.
